tabby-core/src/services/profiles.service.ts

```diff
@@ -211,27 +349,192 @@ export class ProfilesService {
+            builtInGroups.push({
+                id: 'built-in',
+                name: this.translate.instant('Built-in'),
+                editable: false,
+                profiles: [],
+            })
+
+            const ungrouped: PartialProfileGroup<ProfileGroup> = {
+                id: 'ungrouped',
+                name: this.translate.instant('Ungrouped'),
+                editable: false,
+            }
+
+            if (includeProfiles) {
+                for (const profile of profiles.filter(p => p.isBuiltin)) {
+                    let group: PartialProfileGroup<ProfileGroup> | undefined = builtInGroups.find(g => g.id === slugify(profile.group ?? 'built-in'))
+                    if (!group) {
+                        group = {
+                            id: `${slugify(profile.group!)}`,
+                            name: `${profile.group!}`,
+                            editable: false,
+                            profiles: [],
+                        }
+                        builtInGroups.push(group)
+                    }
+
+                    group.profiles!.push(profile)
+                }
+
+                ungrouped.profiles = profiles.filter(p => !p.isBuiltin)
+            }
+
+            groups = groups.concat(builtInGroups)
+            groups.push(ungrouped)
+        }
+
+        return groups
+    }
+
+    /**
+    * Insert a new ProfileGroup in config
+    * arg: saveConfig (default: true) -> invoke after the Profile was updated
+    * arg: genId (default: true) -> generate uuid in before pushing Profile into config
+    */
+    async newProfileGroup (group: PartialProfileGroup<ProfileGroup>, saveConfig = true, genId = true): Promise<void> {
+        if (genId) {
+            group.id = `${uuidv4()}`
+        }
+
+        const cProfileGroup = this.config.store.groups.find(p => p.id === group.id)
+        if (cProfileGroup) {
+            throw new Error(`Cannot insert new ProfileGroup, duplicated Id: ${group.id}`)
+        }
+
+        this.config.store.groups.push(group)
+
+        if (saveConfig) {
+            return this.config.save()
+        }
+    }
+
+    /**
+    * Write a ProfileGroup in config
+    * arg: saveConfig (default: true) -> invoke after the ProfileGroup was updated
+    */
+    async writeProfileGroup (group: PartialProfileGroup<ProfileGroup>, saveConfig = true): Promise<void> {
+        delete group.profiles
+        delete group.editable
+
+        const cGroup = this.config.store.groups.find(g => g.id === group.id)
+        if (cGroup) {
+            Object.assign(cGroup, group)
+
+            if (saveConfig) {
+                return this.config.save()
+            }
+        }
+    }
+
+    /**
+    * Delete a ProfileGroup from config
+    * arg: saveConfig (default: true) -> invoke after the ProfileGroup was deleted
+    */
+    async deleteProfileGroup (group: PartialProfileGroup<ProfileGroup>, saveConfig = true, deleteProfiles = true): Promise<void> {
```

Comment:
please avoid having a saveConfig argument in every function, that's the caller's responsibility

Reply:
Resolved in 06859de2de651867be65b0230b679908b58f2fc6